tests/integration/Microsoft.Bot.Builder.Integration.Runtime.Tests/Components/Implementations/PirateBotComponent.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Bot.Builder.Dialogs.Declarative;
+using Microsoft.Bot.Builder.Dialogs.Declarative.Converters;
+using Microsoft.Bot.Builder.Dialogs.Memory;
+using Microsoft.Bot.Builder.Dialogs.Memory.Scopes;
+using Microsoft.Bot.Builder.Integration.AspNet.Core;
+using Microsoft.Bot.Builder.Integration.Runtime.Tests.Components.TestComponents;
+using Microsoft.Bot.Builder.Runtime.Tests.Components.TestComponents;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Bot.Builder.Runtime.Tests.Components.Implementations
+{
+    public class PirateBotComponent : BotComponent
+    {
+        public override void Startup(
+            IServiceCollection services, 
+            IConfiguration componentConfiguration, 
+            ILogger logger)
+        {
```

Comment:
At some point there was a proposal to add extensions for these 4/5 things:

`services.AddDeclarativeType<T>()`
`services.AddConverterFactory<T>()`
`services.AddMemoryScope<T>()`
`services.AddPathResolver<T>()`
`services.AddBotAdapter<T>()`

Thoughts? I'm OK with the pattern in the code plus documenting it for now and refraining from adding more public surface at the moment, but happy to hear opinions.